contrib/new_post.py

```diff
@@ -0,0 +1,192 @@
+
+    print(f"GitHub PR title: \"{pr.title}\"")
+    print(f"Parsed PR title: {title}")
+    print(f"GitHub PR labels: \"{labels}\"")
+    print(f"Parsed PR labels: \"{components}\"")
+
+    with open(fname, 'w') as f:
+        f.write('---\n')
+        f.write('layout: pr\n')
+        f.write(f"date: {date}\n")
+        f.write(f"title: {title}\n")
+        f.write(f"pr: {pr.number}\n")
+        f.write(f"authors: [{pr.user}]\n")
+        f.write(f"components: {json.dumps(components)}\n")  # Use json.dumps() for double quotes
+        f.write(f"host: {host}\n")
+        f.write("status: upcoming\n")
+        f.write("commit:\n")
+        f.write("---\n\n")
+        f.write("_Notes and questions to follow soon!_\n\n")
+        f.write("<!-- TODO: Before meeting, add notes and questions\n")
+        f.write("## Notes\n\n\n\n")
+        f.write("## Questions\n\n")
+        f.write("1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?\n")
+        f.write("-->\n\n\n")
+        f.write("<!-- TODO: After a meeting, uncomment and add meeting log between the irc tags\n")
+        f.write("## Meeting Log\n\n")
+        f.write("### Meeting 1\n\n")
+        f.write("-->\n")
+        f.write("<!-- TODO: For additional meetings, add the logs to the same irc block. This ensures line numbers keep increasing, avoiding hyperlink conflicts for identical line numbers across meetings.\n\n")
+        f.write("### Meeting 2\n\n")
+        f.write("{% irc %}\n")
+        f.write("{% endirc %}\n")
+        f.write("-->\n")
+
+def load_pr_from_gh(n: int) -> PullRequest:
+    response = urllib.request.urlopen(f"{GITHUB_API_URL}/{n}")
+    data = json.loads(response.read())
+    try:
+        return PullRequest(number=data['number'],
+                           title=data['title'],
+                           labels=[l['name'] for l in data['labels']],
+                           user=data['user']['login'])
+    except KeyError:
+        raise KeyError(f"Could not deserialize GitHub response into PullRequest ({data}")
+
+def main() -> None:
+
+    if sys.version_info < MIN_PYTHON:
+        print(f"Error: This script requires Python {'.'.join(map(str, MIN_PYTHON))} or higher.", file=sys.stderr)
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-d", "--date", help="Meeting date in YYYY-DD-MM format (optional, defaults to next Wednesday)",
+                        type=validate_date, default=next_wednesday())
+    parser.add_argument("-p", "--pr", required=True, type=int, help="PR number (required)")
+    parser.add_argument("-u", "--host", required=True, help="Host's github username (required)")
+
+    args = parser.parse_args()
+
+    # Check file doesn't already exist
+    fname = f"_posts/{args.date}-#{args.pr}.md"
+    if os.path.isfile(fname):
+        sys.exit(f"file {fname} already exists!")
+
```

Comment:
Perhaps useful to add a warning if the PR already exists? Scope creep though so feel free to ignore. (+ `import glob`)

```py
    # Warn if PR has already been covered
    if(previous_meetings := glob.glob(f'_posts/*#{args.pr}*')):
        print(f"WARNING: PR already covered in {previous_meetings[0]}")
```

Reply:
I'll leave this to avoid scope creep. I'm happy to review any future PR though.